File: jobs/2026-05-07__08-46-23/money_transfer_saga_py__Gu4kkdX/artifacts/code/activities.py
from temporalio import activity
import os

LOG_FILE = "/home/user/app/transactions.log"

@activity.defn
async def withdraw(account_id: str, amount: int) -> None:
    with open(LOG_FILE, "a") as f:
        f.write(f"WITHDRAW {account_id} {amount}\n")

@activity.defn
async def deposit(account_id: str, amount: int) -> None:
    if account_id == "B_FAIL":
        raise ValueError("Deposit failed")
    with open(LOG_FILE, "a") as f:
        f.write(f"DEPOSIT {account_id} {amount}\n")

@activity.defn
async def refund(account_id: str, amount: int) -> None:
    with open(LOG_FILE, "a") as f:
        f.write(f"REFUND {account_id} {amount}\n")

File: jobs/2026-05-07__08-46-23/money_transfer_saga_ts__ENbds7M/artifacts/code/src/activities.ts
export async function withdraw(accountId: string, amount: number): Promise<void> {
  console.log(`Withdrawing ${amount} from account ${accountId}`);
}

export async function deposit(accountId: string, amount: number): Promise<void> {
  if (accountId === 'fail') {
    console.error(`Deposit to account ${accountId} failed as requested`);
    throw new Error('Deposit failed');
  }
  console.log(`Depositing ${amount} to account ${accountId}`);
}

export async function refund(accountId: string, amount: number): Promise<void> {
  console.log(`Refunding ${amount} to account ${accountId}`);
}
